### PR TITLE
Set up a "Hello world" project with two apps: RAILS BACK END

### PR DESCRIPTION
**Pull request by JuanDBta**
In this branch you can find the following features:

- Create a new Rails API app called 'hello-rails-back-end'.
- Initialize your project with Git.
- Make sure that your project has a Postgres database set up.
- Create a table for storing your messages and create 5 different greetings.
- Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and rails route).

Here is the link to my [Front End React App PR](https://github.com/JuanDBta/hello-react-front-end/pull/1)

### CODE REVIEW

File: app/controllers/messages_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      redirect_to messages_path
+    else
+      render 'new'
+    end
+  end
+
+  def index
+    @message = Message.order('RANDOM()').first
+    render json: { message: @message.text }
+    puts
```
Comment (brainconnect93) [inline app/controllers/messages_controller.rb:18]: - [ ] Great effort working on your project, but while trying to the directory where you have the messages API endpoint it throws an error and this error makes it difficult to navigate there and all the messages are not shown. Also, this makes the random messages not render on the browser on the front end side. I would kindly suggest you work on it and ensure everything works fine.

Video Recording 👇 


https://github.com/JuanDBta/hello-rails-back-end/assets/78634841/89cb60b2-d5b1-4c68-9a1a-46b9ddc861b9

**Review by brainconnect93 — CHANGES_REQUESTED** [review]
### Hi @JuanDBta, 👋 

# STATUS: Changes Required :recycle:

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Good Points 👍

- Descriptive Pull Request title & summary. ✔️
- "Hello world" project with `react-rails` gem ✔️
- Implement the `API` in the backend correctly ✔️
- Add link to the `Frontend` in the README file ✔️ 
- Good use of Git flow ✔️

### Changes Required

- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @brainconnect93 ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/messages_controller.rb (hunk 1 above)
Comment (JuanDBta) [reply app/controllers/messages_controller.rb:18]: Dear reviewer @brainconnect93:

Thank you for your review. I've checked the app with my partners on their local machines and is working fine.  Maybe you missed something in the database.yml file or running a migration like rails db:create or bin/rails db:seed. This is the URL http://127.0.0.1:3000/messages to see the json messages.

Comment (Felix45) [inline app/controllers/messages_controller.rb:18]: ## Application works as expected
Screenshot from 2023-10-18 01-06-26

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @JuanDBta, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] The two Applications function as expected :white_check_mark: 
- [x] Application uses database `seeder` to populate the database :white_check_mark: 
- [x] Pull request has a descriptive title and summary :white_check_mark: 
- [x] Git flow is used correctly :white_check_mark: 
- [x] Repository has a professional readme file :white_check_mark:     

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/messages_controller.rb (hunk 1 above)
Comment (brainconnect93) [reply app/controllers/messages_controller.rb:18]: @JuanDBta, am so sorry about the mistake, I checked through the whole project again now and I realized the mistake from my machine PGAdmin and now I see the project is working fine.
Am so sorry for the inconvenience my mistake might have caused you. 
@felix, Thanks for approving the project.

**Pull request merged**